### DISCUSSION TITLE
Unable to populate data in "data" field from "msg" field using record transformer

### DISCUSSION THREAD

**Discussion by parshva-shah-dsw** [fluent/fluentd#4466] [Q&A]
certain application are printing logs in "msg" field we are trying to transform the "msg" field to "message" field with record transformer. Whole setup is on kubernetes. below configuration is added in the fluentd configmap. can anyone help here to resolve the issue

References : https://docs.fluentd.org/filter/record_transformer

```xml
<filter kubernetes.**>
    @type record_transformer
     enable_ruby true
     <record>
            message ${if record.has_key?('msg'); then record["msg"]; else "nomsg"; end}
     </record>
</filter>
```

**Comment by daipom** [comment]
Do you mean the code `${if record.has_key?('msg'); then record["msg"]; else "nomsg"; end}` does not work as expected?

I don't understand the expected input and output, but for example, you can do the following.

```xml
<source>
  @type sample
  tag test
  sample {"msg":"AAA"}
</source>

<source>
  @type sample
  tag test
  sample {"message":"BBB"}
</source>

<filter test.**>
  @type record_transformer
  enable_ruby true
  remove_keys msg
  <record>
    message ${record["message"] || record.fetch("msg", "nomsg")}
  </record>
</filter>

<match test.**>
  @type stdout
</match>
```

Output

```
2024-04-16 13:31:06.058144286 +0900 test: {"message":"BBB"}
2024-04-16 13:31:06.059013192 +0900 test: {"message":"AAA"}
...
```
